Sources/AudioKitUI/Visualizations/FFTView.swift

```diff
@@ -22,38 +22,45 @@ class FFTModel: ObservableObject {
         for index in 0..<fftData.count {
             if fftData[index].isNaN { fftData[index] = 0.0 }
         }
+        
+        var tempAmplitudeArray : [Double] = []
 
         // loop by two through all the fft data
         for i in stride(from: 0, to: FFT_SIZE - 1, by: 2) {
-            // get the real and imaginary parts of the complex number
-            let real = fftData[i]
-            let imaginary = fftData[i + 1]
-            
-            let normalizedBinMagnitude = 2.0 * sqrt(real * real + imaginary * imaginary) / Float(FFT_SIZE)
-            let amplitude = Double(20.0 * log10(normalizedBinMagnitude))
             
-            // scale the resulting data
-            var scaledAmplitude = (amplitude + 250) / 229.80
+            if i / 2 < self.amplitudes.count {
             
-            // restrict the range to 0.0 - 1.0
-            if scaledAmplitude < 0 {
-                scaledAmplitude = 0
-            }
-            if scaledAmplitude > 1.0 {
-                scaledAmplitude = 1.0
-            }
-            
-            // add the amplitude to our array (further scaling array to look good in visualizer)
-            DispatchQueue.main.async {
-                if i / 2 < self.amplitudes.count {
-                    var mappedAmplitude = self.map(n: scaledAmplitude, start1: 0.3, stop1: 0.9, start2: 0.0, stop2: 1.0)
-                    if mappedAmplitude < 0.0 {
-                        mappedAmplitude = 0.0
-                    }
-                    self.amplitudes[i / 2] = mappedAmplitude
+                // get the real and imaginary parts of the complex number
+                let real = fftData[i]
+                let imaginary = fftData[i + 1]
+                
+                let normalizedBinMagnitude = 2.0 * sqrt(real * real + imaginary * imaginary) / Float(FFT_SIZE)
+                let amplitude = Double(20.0 * log10(normalizedBinMagnitude))
+                
+                // scale the resulting data
+                var scaledAmplitude = (amplitude + 250) / 229.80
```

Comment:
What's the meaning of these numbers (250 & 229.80)? Were these just tweaked by trial and error? Do they work at different sizes?
Could / should they be parameters of the view?
Thanks for doing this.

Reply:
Thanks for checking out the code. I've been wondering the same thing - I wrote this line awhile back so my memory is a bit fuzzy. I do wonder if we can realize some generalized function here. However, I vaguely remember reading that its common to have to deal with an arbitrary scaling on the output of an fft, but don't quote me on that. I'll look into it and report back.